### PR TITLE
Fix a problem where an user can't update their own 'person' resource.

### PR DESCRIPTION
**Pull request by mworrell**
This fixes issue #1229

### CODE REVIEW

**Comment by ddeboer** [comment]
Persons can now correctly update themselves. This fix breaks manage_schema for ACL rules, however. Will report back when I know more.

File: modules/mod_acl_user_groups/support/acl_user_groups_checks.erl [hunk 1]
```diff
@@ -40,6 +40,7 @@
 -export([
         can_insert_category/2,
         can_insert_category/3,
+        can_update_category/3,
```
Comment (ddeboer) [inline modules/mod_acl_user_groups/support/acl_user_groups_checks.erl]: This function is called nowhere. Is that intentional?

**Comment by ddeboer** [comment]
This worked on 0.x but fails on this branch:

```erlang
{ok, Id} = m_rsc:insert([{category_id, page_menu}], z_acl:sudo(C)).
m_rsc:update(Id, [{title, "iets"}], z_acl:sudo(C)).
** exception throw: {{error,eacces},
                     [{m_rsc_update,update_transaction_fun_props,3,
                                    [{file,"src/models/m_rsc_update.erl"},{line,365}]},
                      {z_db,'-transaction1/2-fun-0-',3,
                            [{file,"src/db/z_db.erl"},{line,141}]},
                      {timer,tc,2,[{file,"timer.erl"},{line,179}]},
                      {z_db,with_connection,3,
                            [{file,"src/db/z_db.erl"},{line,206}]},
                      {z_db,transaction,3,[{file,"src/db/z_db.erl"},{line,95}]},
                      {m_rsc_update,update_transaction,3,
                                    [{file,"src/models/m_rsc_update.erl"},{line,303}]},
                      {erl_eval,do_apply,6,[{file,"erl_eval.erl"},{line,661}]},
                      {shell,exprs,7,[{file,"shell.erl"},{line,684}]}]}
     in function  m_rsc_update:update_result/3 (src/models/m_rsc_update.erl, line 349)
```

**Comment by ddeboer** [comment]
By the way, we really need functional tests (#1235) to automatically detect regressions like these.

**Comment by mworrell** [comment]
@ddeboer the `observe_acl_rsc_update_check` can also restrict properties and fill in defaults. It is also run inside the update transaction. It is a last stage check before the actual update.

**Comment by ddeboer** [comment]
Thanks for the fix. 

Currently, `observe_acl_rsc_update_check` duplicates a lot of checks that are also done in `observe_is_allowed`, which increases the potential for bugs (as we saw here). Do we need to do the same checks again or can we re-use the logic from `is_allowed`?

**Comment by mworrell** [comment]
Maybe we can refactor, but let's do that in another issue, after the collaboration groups are merged as well.

**Comment by ddeboer** [comment]
Fair enough.

**Pull request merged**